libraries/AP_HAL_Linux/Util.cpp

```diff
@@ -174,33 +176,25 @@ const char *Linux::Util::_hw_names[UTIL_NUM_HARDWARES] = {
 #define MAX_SIZE_LINE 50
 int Util::get_hw_arm32()
 {
-    int ret = -ENOENT;
-    char buffer[MAX_SIZE_LINE];
-    const char* hardware_description_entry = "Hardware";
-    char* flag;
-    FILE* f;
-
-    f = fopen("/proc/cpuinfo", "r");
-
+    char buffer[MAX_SIZE_LINE] = { 0 };
+    FILE *f = fopen("/proc/cpuinfo", "r");
     if (f == NULL) {
-        ret = -errno;
-        goto end;
+        return -errno;
     }
 
     while (fgets(buffer, MAX_SIZE_LINE, f) != NULL) {
-        flag = strstr(buffer, hardware_description_entry);
-        if (flag != NULL) {
-            for (uint8_t i = 0; i < UTIL_NUM_HARDWARES; i++) {
-                if (strstr(buffer, _hw_names[i]) != 0) {
-                     ret = i;
-                     goto close_end;
-                }
+        if (strstr(buffer, "Hardware") == NULL) {
+            continue;
+        }
+        for (uint8_t i = 0; i < UTIL_NUM_HARDWARES; i++) {
+            if (strstr(buffer, _hw_names[i]) == 0) {
+                continue;
             }
+            fclose(f);
+            return i;
         }
     }
 
-close_end:
```

Comment:
I'd still prefer using the labels, but I think it doesn't hurt removing them in this case.